lib/web/app/handler.go

```diff
@@ -145,6 +238,22 @@ func NewHandler(ctx context.Context, c *HandlerConfig) (*Handler, error) {
 	return h, nil
 }
 
+// appForPublicEndpoint returns an App given a public endpoint.
+// If no app is found for the given public endpoint, trace.NotFound is returned.
+func (h *Handler) appForPublicEndpoint(ctx context.Context, publicAddr string) (types.Application, error) {
+	allAppServers, err := h.c.AccessPoint.GetApplicationServers(ctx, defaults.Namespace)
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+	publicAddressMatcher := MatchPublicAddr(publicAddr)
+	for _, a := range allAppServers {
+		if publicAddressMatcher(ctx, a) {
```

Comment:
Maybe:
```suggestion
	publicAddressMatches := MatchPublicAddr(publicAddr)
	for _, a := range allAppServers {
		if publicAddressMatches(ctx, a) {
```
It will read more naturally